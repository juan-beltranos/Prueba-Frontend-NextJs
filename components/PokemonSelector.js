import { Form } from 'react-bootstrap';

const PokemonSelector = ({ handleTypeChange }) => {
  return (
    <Form.Group controlId="">
      <Form.Label>Selecciona un elemento:</Form.Label>
      <Form.Control as="select" onChange={handleTypeChange} >
        <option value="">Todos los elementos</option>
        <option value="fire">Fuego</option>
        <option value="water">Agua</option>
        <option value="ground">Tierra</option>
      </Form.Control>
    </Form.Group>
  );
};

export default PokemonSelector;
