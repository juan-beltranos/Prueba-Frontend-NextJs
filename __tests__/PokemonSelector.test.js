import renderer from 'react-test-renderer';
import PokemonSelector from '../components/PokemonSelector';

it('selected Pokemons state', () => {
    const element = 'fire'
    const component = renderer.create(
        <PokemonSelector handleTypeChange={element} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});