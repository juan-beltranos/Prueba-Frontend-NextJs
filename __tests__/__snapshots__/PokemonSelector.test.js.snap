// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`selected Pokemons state 1`] = `
<div
  className="form-group"
>
  <label
    className="fw-bold fs-2 form-label"
  >
    Selecciona un elemento y escoge tu Pok√©mon:
  </label>
  <select
    className="shadow bg-white rounded-3 fs-4 form-control"
    onChange="fire"
  >
    <option
      value=""
    >
      Todos los elementos
    </option>
    <option
      value="fire"
    >
      Fuego
    </option>
    <option
      value="water"
    >
      Agua
    </option>
    <option
      value="ground"
    >
      Tierra
    </option>
    <option
      value="electric"
    >
      Electrico
    </option>
    <option
      value="ice"
    >
      Hielo
    </option>
  </select>
</div>
`;
